src/PostgREST/ApiRequest.hs

```diff
@@ -152,6 +157,7 @@ userApiRequest schema req reqBody =
   , iPreferRepresentation = representation
   , iPreferSingular = singular
   , iPreferCount = not $ singular || hasPrefer "count=none"
+  , iProxy = pickProxy $ lookupHeader "x-forwarded-uri"
```

Comment:
since this is used only for openAPI, how about string here only the raw value of the header and call the pickProxy only when generating the openapi output, and while doing that ... move all these new functions in the openapi module